src/lib/logic/typing/globals/contexts.ts

```diff
@@ -0,0 +1,37 @@
+import type { Writable } from "svelte/store";
+import type { Errors } from "../utils/errors.js";
+import type {
+  ActionConfig,
+  Submit,
+  SubmitAction,
+  SubmitOptions
+} from "../stores/form.js";
+import type {
+  ErrorsStyles,
+  FileStyles,
+  Icons,
+  FieldStyles,
+  OptionStyles,
+  SelectStyles
+} from "./styles.js";
+
+export interface ContextStyles {
+  field?: FieldStyles;
+  file?: FileStyles;
+  select?: SelectStyles;
+  option?: OptionStyles;
+  icons?: Icons | null;
+  errors?: ErrorsStyles;
+}
+
+export interface ContextForm<Values, Fields> {
+  loading: Writable<boolean>;
+  errors: Writable<Errors>;
+  data: Writable<Values>;
+  setError(name: Fields, error?: unknown): void;
+  setField(field: Fields, value: unknown, validate?: boolean): Promise<void>;
+  check(event: FocusEvent | Event): Promise<void>;
+  action(data?: ActionConfig): Promise<void>;
+  submit<T extends Values>(action: SubmitAction<T>, options?: SubmitOptions): Submit;
+  styles: Writable<ContextStyles>;
+}
```

Comment:
🚫 **[eslint]** <[eol-last](https://eslint.org/docs/latest/rules/eol-last)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Newline required at end of file but not found.
```suggestion
}

```